File: Clients/mobile courier/app/src/main/java/com/nomnomgo/courier/models/AuthModels.kt
package com.nomnomgo.courier.models

import com.google.gson.annotations.SerializedName

// Модели для авторизации
data class LoginRequest(
    val login: String,
    val password: String
)

data class RegisterRequest(
    val email: String,
    val username: String,
    val password: String,
    val phoneNumber: String? = null,
    val isCourier: Boolean = true
)

data class LoginResponse(
    val userId: String? = null, // Сделать nullable, т.к. сервер не возвращает это поле
    val username: String,
    val accessToken: String,
    val refreshToken: String,
    val expiresAt: String
)

data class RegisterResponse(
    val userId: String? = null, // Сделать nullable, т.к. сервер не возвращает это поле
    val username: String,
    val accessToken: String,
    val refreshToken: String,
    val expiresAt: String
)
data class RefreshTokenRequest(
    val refreshToken: String
)

data class RefreshTokenResponse(
    val accessToken: String,
    val refreshToken: String,
    val expiresAt: String
)

data class CurrentUserResponse(
    val userId: String,
    val username: String,
    val email: String,
    val phoneNumber: String?,
    val roles: List<String>
)

data class OrderItem(
    val name: String,
    val quantity: Int,
    val price: Double
)

// Модели для обновления статуса
data class CourierStatusUpdate(
    val courierId: String,
    val orderId: String,
    val status: String // "Ready", "InDelivery", "Delivered"
)

// Модели для геолокации
data class LocationUpdate(
    val courierId: String,
    val latitude: Double,
    val longitude: Double,
    val timestamp: Long = System.currentTimeMillis()
)

data class ApiError(
    val error: String
)

File: Clients/Frontend/nomnom-frontend/src/app/core/models/auth.models.ts
// src/app/core/models/auth.models.ts

export interface RegisterRequest {
  email: string;
  username: string;
  password: string;
  phoneNumber?: string;
}

export interface LoginRequest {
  login: string;
  password: string;
}

export interface RefreshTokenRequest {
  refreshToken: string;
}

// Responses
export interface CurrentUserResponse {
  userId: string;
  username: string;
  email: string;
  phoneNumber?: string;
  isBlocked: boolean;
  blockedUntil?: string;
  roles: string[];
  permissions: string[];
  createdAt: string;
}

export interface LoginResponse {
  accessToken: string;
  refreshToken: string;
  username: string;
  roles: string[];
  expiresAt: string;
}

export interface RegisterResponse {
  userId: string;
  username: string;
  accessToken: string;
  refreshToken: string;
  expiresAt: string;
}

export interface RefreshTokenResponse {
  accessToken: string;
  refreshToken: string;
  expiresAt: string;
}

// Интерфейс для локального использования (совместимость)
export interface CurrentUser {
  userId: string;
  username: string;
  email: string;
  phoneNumber?: string;
  roles: string[];
  createdAt: string;
}
